### PR TITLE
feat: add basic functionality for icicle chart to display static data

### PR DESCRIPTION
**Pull request by cguan7**
🏆 Enhancements
adds functionality for the icicle chart to have data passed in and display (static, no interactions
yet)

### CODE REVIEW

**Comment by williaster** [comment]
do you have any screenshots?

File: packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx [hunk 1]
```diff
@@ -50,10 +79,82 @@ export default class IcicleEventChart extends Component<Props> {
...
+    const { boxMargin, color, contentRenderer, isVertical, width, height, rounding } = this.props;
+
+    const root = this.props.d3TreeRoot;
+
+    // Clear all elements and redraw the Icicle Chart
+    d3Select(this.chartRef.current)
+      .selectAll('*')
+      .remove();
+
+    const svg = d3Select(this.chartRef.current)
```
Comment (williaster) [inline packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx:109]: I'm curious why you aren't using a library like [`@vx` ](https://vx-demo.now.sh/gallery) that plays better with `react` and renders visualizations/utilizes `d3` more declaratively.

**Comment by cguan7** [comment]
Yes, I have some some screenshots! This is what it looks like currently in Storybook with the intention that it is not the final look. Since this PR was already getting a little big, I wanted to make it in a separate one:

image

Additionally, the README for this package has a screenshot of a prototype that is slightly outdated (we're still going through design, but have made some changes to spacing between icicles, etc.)

File: packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx (hunk 1 above)
Comment (cguan7) [reply packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx:109]: I have no good answer for that other than being not very familiar with vx... I recall seeing it in the `event-flow` viz if I remember correctly but didn't look too much into it. Will definitely look more into that library tomorrow in the morning!

File: packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx [hunk 2]
```diff
@@ -29,14 +31,41 @@ interface Props {
     y: number;
   };
   color: (name: string) => string;
-  contentRenderer: () => void;
+  contentRenderer: (...args: any[]) => void;
```
Comment (khtruong) [inline packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx]: Can we more more explicit on the type?

File: packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx (hunk 1 above)
Comment (khtruong) [reply packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx:109]: Very good point! The visualization will have some animation as well and it doesn't look like vx supports it. Although we could use other react animation libraries with vx. I think given Chris's deadlines, I suggest that this is something we look into later on. I am not a big fan of how unreadable d3 either but fortunately, I don't think there is much more than what is already in this PR.

File: packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx (hunk 2 above)
Comment (cguan7) [reply packages/superset-ui-plugin-chart-icicle-event/src/IcicleEventChart.tsx]: Updated to include the types, `BaseType` is documented here:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/d3-selection/index.d.ts#L21

**Review by khtruong — APPROVED** [review]

**Pull request merged**